mesonbuild/backend/ninjabackend.py

```diff
@@ -1032,6 +1032,14 @@ def generate_target(self, target):
         elem = self.generate_link(target, outname, final_obj_list, linker, pch_objects, stdlib_args=stdlib_args)
         self.generate_dependency_scan_target(target, compiled_sources, source2object, generated_source_files, fortran_order_deps)
         self.add_build(elem)
+        #In AIX, we archive shared library. If the instance is a shared library we add a command to archive the shared library
+        #object and create the build element.
+        if mesonlib.is_aix() and isinstance(target, build.SharedLibrary):
```

Comment:
This is wrong. You need to check if the host_machine is AIX right?

Reply:
not host_machine, you need to do somethg like `self.environment.machines[target.for_machine].is_aix()` (I can't remember off the top of my head where environment is in the backend

Reply:
I have used this self.environment.machines[target.for_machine].is_aix() in my latest commit in this PR. Kindly check